### ISSUE TITLE
Leading slash on filepath breaks SSL as it cannot find the cert / key.

### ISSUE THREAD

**Issue by Aterfax** [danielperna84/hass-configurator#117]
In short, using the leading slash to denote the root of the file system (i.e. giving it an absolute path with leading slash doesn't seem to work.)

e.g. the former is broken and will not start

  "certfile": "/ssl/letsencrypt/fullchain.pem",
  "keyfile": "/ssl/letsencrypt/privkey.pem",

The below however does work:

  "certfile": "ssl/letsencrypt/fullchain.pem",
  "keyfile": "ssl/letsencrypt/privkey.pem",

Running on Hassio - configurator version is the most up to date I assume.

Settings options are as follows:

```
{
  "username": "admin",
  "password": "PASSWORD",
  "ssl": true,
  "certfile": "ssl/letsencrypt/fullchain.pem",
  "keyfile": "ssl/letsencrypt/privkey.pem",
  "allowed_networks": [
    "10.0.0.0/16",
    "172.30.0.0/16"
  ],
  "banned_ips": [
    "8.8.8.8"
  ],
  "banlimit": 0,
  "ignore_pattern": [
    "__pycache__"
  ],
  "dirsfirst": false,
  "enforce_basepath": false,
  "notify_service": "persistent_notification.create"
}
```

**Comment by danielperna84** [comment]
The certificate path is hardcoded to `ssl/` [here](https://github.com/home-assistant/hassio-addons/blob/54e762542050daccf427f7a35d994bcf47ce1aa6/configurator/run.sh#L26). It's been like that most of the time, except a few weeks ago where I broke that while updating the Dockerfile. But right now it is like it has always been. So regular hassio users who have their certificates in the ssl-directory won't have to do anything.
Is there a particular reason why your paths differ from other hassio-users?

**Comment by Aterfax** [comment]
Yes, I have a few devices in need of SSL certificates which are all grabbed on a central server and then shared and remounted over NFS. I cannot mount the /ssl directory itself as Hassio will throw an error so I mount it to a subdirectory.

Ultimately another quick solution is to simply use a relative symbolic link in the ssl directory to point at the sub-directory and certificates / keys within matching the standard Hassio file names.

Maybe some documentation that absolute paths won't work  / it needs relative paths from the  /ssl directory?

**Comment by danielperna84** [comment]
> Maybe some documentation that absolute paths won't work / it needs relative paths from the /ssl directory?

Where would you expect this information to be? It's not directly related to the configurator since the other addons that use SSL probably to is exactly like the configurator. So in my opinion this should be a general statement about SSL-setups with hassio.

**Comment by Aterfax** [comment]
I would personally document it at the Hassio configurator level and again at the plugin level because developers can modify how they interpret the field entry.

Former certainly if modified.

**Comment by danielperna84** [comment]
You mean [here](https://www.home-assistant.io/addons/configurator/)? Then [this](https://github.com/home-assistant/home-assistant.io/blob/next/source/_addons/configurator.markdown) would need the file that needs to be changed.
Since I don't use hassio I don't know where to other correct place would be (never really looked at the documentation). In any case, you can fork the repository I linked to above, make the changes you think are helpful to others, and then open a pull request.

**Comment by Aterfax** [comment]
Not necessarily on the configurator documentation - I just meant at that level. The SSL field descriptions probably need updating on that page to reflect the /ssl path anyway and make it clear how it expects to see the files and how most plugins will also expect to see them too.

 It could be fudged into there but users probably wouldn't expect to find that info there for plugins?

 I'm not sure there is a master "plugin page" which dicusses the fields and their conditions - as yet I only saw documentation per plugin which I why I was recommending it for each plugin documentation and making it explicit for all of them.

**Comment by danielperna84** [comment]
> So it probably needs pushing up across all documentation to be unified.

That's why I'm pointing to the documentation-repository. 😉
